components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -111,6 +112,19 @@ private void insertAccessToken(String accessToken, String consumerKey, AccessTok
                     "Authorized user should be available for further execution.");
         }
 
+        if (OAuthServerConfiguration.getInstance().usePersistedAccessTokenAlias()) {
+            try {
+                accessToken = OAuthServerConfiguration.getInstance().getIdentityOauthTokenIssuer()
+                        .getAccessTokenHash(accessToken);
+            } catch (OAuthSystemException e) {
+                if (log.isDebugEnabled() &&
+                        IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.ACCESS_TOKEN)) {
+                    log.debug("Error while getting access token hash for token: " + accessToken);
```

Comment:
Even though we check isTokenLoggable, better if we log the access token as a hash. Not the direct value. Please check how we did it in other places.